### PR TITLE
Add UsageService class, specs, and related changes.

### PR DESCRIPTION
**Pull request by djberg96**
This PR adds a UsageService class that people (including us) can potentially use to track resource consumption.

https://msdn.microsoft.com/en-us/library/azure/mt219001

### CODE REVIEW

**Comment by djberg96** [comment]
@bzwei Please review.

File: lib/azure/armrest/billing/usage_service.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
+require 'cgi'
```
Comment (bzwei) [inline lib/azure/armrest/billing/usage_service.rb]: where is this library used?

File: lib/azure/armrest/billing/usage_service.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+module Azure
+  module Armrest
+    module Billing
+      class UsageService < ArmrestService
+        # Creates and returns a new UsageService object.
+        #
+        def initialize(configuration, options = {})
+          options['api_version'] = '2015-06-01-preview' # Must hard code for now
```
Comment (bzwei) [inline lib/azure/armrest/billing/usage_service.rb]: you may want to do a `merge`, leaving the argument `options` unmodified

File: lib/azure/armrest/billing/usage_service.rb (hunk 1 above)
Comment (djberg96) [inline lib/azure/armrest/billing/usage_service.rb]: Ah, forgot to remove it. Was thinking I had to encode the dates, but rest-client already does it.

**Comment by djberg96** [comment]
@bzwei I've cleaned it up.

**Pull request merged**